### DISCUSSION TITLE
Auto-delete depleted spools + Filament switch question

### DISCUSSION THREAD

**Discussion by NevesLF** [Donkie/Spoolman#272] [Ideas]
1) Hi, sorry if that's already a feature but I couldn't find it. Is there a way to automatically delete a spool from Spoolman once it reaches 0 g? If not, could something like this be added in the future?

2) Unrelated (probably silly) question: I've seem someone mention Spoolman supports multimaterial, so I'd imagine it also supports filament switch on regular printers ("pause at height" to manually switch filaments), but how would that work? Can I simply select a different filament on Spoolman once the printer is paused and it will work from there?

**Comment by bofh69** [comment]
1. I'm not sure if it is such a good idea. Many manufacturers add some more filament on the spool than what they say on the label so it would then be common for the spool to be removed before it is fully used up, if everything is perfectly calibrated. Usually things are not perfectly calibrated either so you can either run out of filament before it is used up in Spoolman, or vice versa.

3. Its Moonraker that knows about the active spool, Spoolman doesn't really know about it. That's why you can use Spoolman with multiple printers. If you have a call to SET_ACTIVE_SPOOL with the new spool's ID in the new filament's custom start gcode, klipper will tell Moonraker about the new spool and it will just work. If not, you should switch the spool in Mainsail/fluidd so they'll tell Moonraker about the new spool.

**Reply by spyder007** [reply]
Regarding your first point, I recently made some changes that should let you track spools using their exact weight, based on a measured weight and the empty spool weight.  However, I would agree that "auto deleting an empty spool" is probably not great.  But, since there is the archive feature, I'd definitely auto-archive empties, but would want that to be a setting that could be toggled.

As to the second point, I noticed a number of requests around integrations for specific programs, but the maintainer has done a great job of keeping spoolman to be what it really is:  an inventory management software for your filament.  Any "integrations" between thinks like Moonraker should be more on the moonraker side, IE, moonraker should report usage to Spoolman, based on which spools it knows it's using.  The trick is, how can Moonraker find the spool it's using in Spoolman?

I am currently using a Bambu P1S and Bambu studio, and I can't seem to find a way to even write a plugin for Bambu studio to let it figure that out.  I'm going to look around a little more, but I  have a hunch Home Assistant and NodeRed are going to play a roll in keeping spoolman up to date.

**Reply by mtinman** [reply]
AFAIK, Bambu Lab's biggest disadvantage of the [Automated Material System (AMS)](https://all3dp.com/2/bambu-lab-ams-simply-explained/) is the fact that it’s closed-source and the only people that can add plugins to/for AMS are the people at Bambu themselves.
